### PR TITLE
AddTripView and TripListView

### PR DESCRIPTION
**Pull request by SamYang10086**

### CODE REVIEW

**Review by SamYang10086 — COMMENTED** [review]
wait for review

File: Travel-Planner/Travel-Planner/Views/TripListView/TripCardView.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                    .padding(.top, 20)
+                    .padding(.leading, 20)
+                Spacer()
+            }
+            .frame(height: 150)
+        }
+        .frame(height: 150)
+        .cornerRadius(15)
+        .shadow(radius: 5)
+        .padding(.vertical, 5)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Views/TripListView/TripCardView.swift]: Is this the padding for list cell view? make the gate narrower

File: Travel-Planner/Travel-Planner/ViewModels/TripViewModel.swift [hunk 2]
```diff
@@ -39,6 +39,9 @@ class TripViewModel: ObservableObject {
         newTrip.endDate = endDate
         newTrip.createdAt = Date()
         
+        let defaultImages = ["tripListImage1", "tripListImage2","tripListImage3","tripListImage4", "tripListImage5","tripListImage6", "tripListImage7","tripListImage8","tripListImage9", "tripListImage10","tripListImage11", "tripListImage12","tripListImage13","tripListImage14", "tripListImage15"]
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/ViewModels/TripViewModel.swift]: move static variable to the top

File: Travel-Planner/Travel-Planner/Utilities/DateUtilities.swift [hunk 3]
```diff
@@ -51,4 +51,10 @@ struct DateUtils {
                                                   month: components.month,
                                                   day: components.day)) ?? date
     }
+    
+    static func formattedDateRange(start: Date, end: Date) -> String {
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Utilities/DateUtilities.swift:56]: give common of what it does. And give a example of the output

File: Travel-Planner/Travel-Planner/Screens/TripListView.swift [hunk 4]
```diff
@@ -10,37 +10,54 @@ import SwiftUI
 struct TripListView: View {
     @StateObject var vm = TripViewModel()
     @State private var showAddTrip: Bool = false
-    
+    @Environment(\.editMode) var editMode
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripListView.swift]: Do we still need this? If not, remove it

File: Travel-Planner/Travel-Planner/Screens/TripListView.swift [hunk 5]
```diff
@@ -10,37 +10,54 @@ import SwiftUI
...
-                        }
+                .onDelete { indices in
+                    for index in indices {
+                        let trip = vm.trips[index]
+                        vm.deleteTrip(trip: trip)
                     }
                 }
             }
-            .onAppear() {
-                // getTrips on Apear
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripListView.swift:36]: Remove this cause a bug that the trips not display

**Review by Topia99 — COMMENTED** [review]
good! need some small fix

File: Travel-Planner/Travel-Planner/Screens/TripListView.swift [hunk 6]
```diff
@@ -10,37 +10,54 @@ import SwiftUI
...
-            ScrollView{
-                LazyVStack {
-                    ForEach(vm.trips) { trip in
+            List {
+                ForEach(vm.trips, id: \.self) { trip in
+                    ZStack(alignment: .leading) {
+                        // Apply horizontal padding to create space on both sides
+                        TripCardView(trip: trip)
+                            .padding(.vertical, 5)
+                            .padding(.horizontal, 16) // Add horizontal padding
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripListView.swift]: make it default padding like "padding(.horizontal)

**Review by SamYang10086 — COMMENTED** [review]
good

**Review by Topia99 — APPROVED** [review]

**Pull request merged**